File: src/scala/FibonacciIterative.scala
object FibonacciIterative {
  def fibonacci(n: Int): BigInt = {
    if (n <= 1) {
      return n
    }

    var a = BigInt(0)
    var b = BigInt(1)
    var result = BigInt(0)

    for (i <- 2 to n) {
      result = a + b
      a = b
      b = result
    }

    result
  }

  def main(args: Array[String]): Unit = {
    val index = 15
    println(s"Fibonacci (iterative) of $index is: ${fibonacci(index)}")
  }
}

File: src/javascript/FibonacciIterative.js
function fibonacci(n) {
  if (n <= 1) {
    return n;
  }

  let a = 0;
  let b = 1;
  let result;

  for (let i = 2; i <= n; i++) {
    result = a + b;
    a = b;
    b = result;
  }

  return result;
}

function main() {
  console.log('Fibonacci (iterative) of 30:', fibonacci(30));
}

main();
